### PR TITLE
Implement default welcome page and allow custom URL /w config

### PR DESCRIPTION
**Pull request by lukebarnard1**
counterpart to https://github.com/matrix-org/matrix-react-sdk/pull/922

### CODE REVIEW

File: config.sample.json [hunk 1]
```diff
@@ -11,5 +11,6 @@
             "matrix.org"
         ]
     },
-    "welcomeUserId": "@RiotBot:matrix.org"
+    "welcomeUserId": "@RiotBot:matrix.org",
+    "welcomePageUrl": "https://about.riot.im"
```
Comment (dbkr) [inline config.sample.json]: doesn't this want to be home.html?

File: src/components/structures/HomePage.js [hunk 2]
```diff
@@ -25,15 +25,24 @@ module.exports = React.createClass({
     displayName: 'HomePage',
 
     propTypes: {
+        // URL base of the team server.
         teamServerUrl: React.PropTypes.string.isRequired,
-        teamToken: React.PropTypes.string.isRequired,
-        collapsedRhs: React.PropTypes.bool,
+        // Team token. Optional. If unset, the homePageUrl will be used
```
Comment (dbkr) [inline src/components/structures/HomePage.js]: url will be used as the team token?

File: config.sample.json (hunk 1 above)
Comment (dbkr) [inline config.sample.json]: Oh I see, home.html is the default if one isn't defined. I think this is going to provide quite strange behaviour for people who just copy the sample config assuming the values in it are sane (ie. a lot of people probably)

File: res/home.html [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Riot - Home</title>
+</head>
+<body>
+    <div>Welcome to Riot</div>
+</body>
+</html>
```
Comment (dbkr) [inline res/home.html]: What's the long-term intention here? Presumably not a minimal, css-less page?

File: src/components/structures/HomePage.js (hunk 2 above)
Comment (dbkr) [inline src/components/structures/HomePage.js]: Oh, I see what's going on. Comment could probably make this clearer. :)

File: config.sample.json (hunk 1 above)
Comment (lukebarnard1) [inline config.sample.json]: Fair enough, I'll just remove it

File: res/home.html (hunk 3 above)
Comment (lukebarnard1) [inline res/home.html]: Tom and Ben are working on it and we'll have something before `new-guest-access` goes live

**Pull request merged**